controller/config.go

```diff
@@ -281,6 +281,14 @@ const (
 
 	// IdleConnectionTimeout is the time between the controller resetting all idle connections.
 	IdleConnectionTimeout = "idle-connection-timeout"
+
+	// ReadTimeout is the maximum duration for reading the entire HTTP request, including the body.
+	// A zero value means no timeout.
+	ReadTimeout = "read-timeout"
```

Comment:
```go
    // HTTPServerReadTimeout is the maximum duration for reading the entire HTTP request, 
    // including the body.
	// A zero value means no timeout.
    HTTPServerReadTimeout = "http-server-read-timeout"`
```
And similar for the `Write` variant below. Current names are too vague.

Reply:
Great idea. Updated on https://github.com/juju/juju/pull/21434/commits/02dc7bab4054d3f50aab90c1c963c6f59e986bc6.